task_processing/plugins/kubernetes/kubernetes_pod_executor.py

```diff
@@ -420,6 +451,37 @@ def _pending_event_processing_loop(self) -> None:
 
         logger.debug("Exiting Pod event processing - stop requested.")
 
+    def _reconcile_task_loop(self) -> None:
+        """
+        Run in a thread to reconcile task_metadata from k8s.
+        """
+        logger.info(f'Waiting {self.refresh_reconciliation_thread_grace}s before doing work')
+        sleep(self.refresh_reconciliation_thread_grace)
+        logger.debug("Starting Pod task config reconciliation.")
+        while not self.stopping:
+            try:
+                # fetch all pods in the target namespace in one request so that we
+                # don't block on making N serial requests to the Kubernetes API
+                pods = self.kube_client.get_pods(namespace=self.namespace)
+            except Exception:
+                logger.exception(
+                    f"Hit an exception attempting to fetch pods in namespace {self.namespace}")
+                pods = None
+                continue
```

Comment:
Because we're `continue`ing here before the sleep interval, if something is wrong with either the api or tron's connection to the api, this seems like it could get us into a constant infinite loop of attempting to fetch the pods w/o pauses in between, which could overwhelm the k8s api (or even this reconciliation thread?). 

We might want to avoid that and add some backoff here if we hit an exception.

Reply:
hm, I'm not sure why I added this continue here. I don't think we actually need it